### ISSUE TITLE
Negative rate constants

### ISSUE THREAD

**Issue by jrfaeder** [RuleWorld/bionetgen#189]
BNG currently does not issue warning or error if rate constants have negative values. This is probably undesirable.

Example here:

```
begin model
begin parameters
    # Cell volume
    k1 1
    k2 -1
    k3 0
    k4 1
    k5 1
    k6 0
end parameters 
begin molecule types
    A()
    B()
    C()
end molecule types
begin observables
Molecules A A()
Molecules B B()
Molecules C C()
end observables
begin reaction rules
    0 -> A()  10^(k1)-1
    A()-> B() -0.5
    B()-> C() 10^(k3)-1
    #A()-> C() 10^(k4)-1
    A()+C() -> C() 10^(k5)-1
    A() -> 0 10^(k6)-1
end reaction rules
end model
generate_network({})
writeMfile({})
simulate({method=>"ode", t_end=>1,n_steps=>20,atol=>1.0E-08})
```
